torch/distributed/fsdp/_exec_order_utils.py

```diff
@@ -147,11 +143,11 @@ def record_post_forward(self, handle: Optional[FlatParamHandle]) -> None:
         if not handle:
             return
         # Only record the first usage of a handles key
-        if handle in self.handles_to_post_forward_order_index:
+        if handles_key._post_forward_index:
```

Comment:
I am worried about the `_post_forward_index = 0` case:
```suggestion
        if handles_key._post_forward_index is not None:
```